CXEC-20507: Add Cypress E2E scenarios for viewing ticket details via URL

ishrarhussain opened the pull request.

RobertOzga87 committed:
fix: make e2e test for CDS Merchandising green (#16043) CXCDS-5231

* CXCDS-5231 fixing e2e tests, part1 search phrase fixes

* CXCDS-5231 fixing e2e tests, part2 clickstreamEvents

* CXCDS-5231 fixing tests, part3

* CXCDS-5231 fixing code style

* CXCDS-5231 formatting fixes

* CXCDS-5231 removal of dead code

* CXCDS-5231 - cleanup, migration documentation and scripts

* Trigger Build

Co-authored-by: Robert Ozga <robert.ozga@sap.com>
Co-authored-by: Kinga Karlińska <kinga.karlinska@sap.com>
Co-authored-by: Patryk Smul <psmul@divante.pl>

steinsebastian committed:
feat: sticky group headers on configurator overview page (#15933)

ChristophHi committed:
chore: reduce complexity for configurator (#16078)

Address sonar findings on cognitive complexity

dependabot[bot] committed:
chore(deps): bump @actions/core in /.github/api-extractor-action (#16100)

Bumps [@actions/core](https://github.com/actions/toolkit/tree/HEAD/packages/core) from 1.2.6 to 1.9.1.
- [Release notes](https://github.com/actions/toolkit/releases)
- [Changelog](https://github.com/actions/toolkit/blob/main/packages/core/RELEASES.md)
- [Commits](https://github.com/actions/toolkit/commits/HEAD/packages/core)

---
updated-dependencies:
- dependency-name: "@actions/core"
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump @actions/core from 1.2.6 to 1.9.1 in /.github/cache-builded-libs (#16099)

* chore(deps): bump @actions/core in /.github/cache-builded-libs

Bumps [@actions/core](https://github.com/actions/toolkit/tree/HEAD/packages/core) from 1.2.6 to 1.9.1.
- [Release notes](https://github.com/actions/toolkit/releases)
- [Changelog](https://github.com/actions/toolkit/blob/main/packages/core/RELEASES.md)
- [Commits](https://github.com/actions/toolkit/commits/HEAD/packages/core)

---
updated-dependencies:
- dependency-name: "@actions/core"
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* force rerun

* force rerun

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <giancarlo.cordero.ortiz@sap.com>

bgambocjaviniar committed:
chore: Cater to other pipelines for Cypress parallelism and creation of junit xml reports for SAP pipeline (#16090)

closes https://jira.tools.sap/browse/CXSPA-867

FollowTheFlo committed:
chore: add unit tests xml reporting (#16048)

SAP Code compliance FC-1 and FC-3 require unit test results to be available in XML format.
karma-junit-reporter library has been added to generate Unit test XML reports.
XML reports are generated in 'unit-tests-reports' folder with distinct file names, eg:unit-test-product.xml, unit-test-qualtrics.xml...
Unit test ci-scripts have been modified to be suitable for all pipelines: custom output stream (dev/fd/5) has been removed.

kimhw0630 committed:
feat: add purchase order column to order history list (#15885)

added purchase order and Cost Center columns to the order history list table
BREAKING CHANGE:
two new columns: PONumber and costCenter

bgambocjaviniar committed:
chore: Make base branches that runs the pipeline to have distinct values (#16124)

fixes BUILD_NUMBER on 'base' merged commits

kimhw0630 committed:
feat: update md file (#16127)

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
fix: AMP violation on Replenishment Orders Page (#16038)

* fix: AMP violation on Replenishment Orders Page

* fix unit test

* updated breaking change

* updated breaking change doc

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
fix: AMP violations on Approval dashboard Page (#16045)

* fix: AMP violations on Approval dashboard Page

* updated breaking change

* updated breaking change doc

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

bgambocjaviniar committed:
chore: Make cypress have 1 run on the dashboard instead of separate runs (#16136)

closes https://jira.tools.sap/browse/CXSPA-1274

bullfroggy-sap committed:
Feat: Add handling for remove voucher events (#15937)

* Add handling for remove voucher events

bullfroggy-sap committed:
Feat: Add handling for add voucher events (#15936)

* Add handling for add voucher events

gladius-mtl committed:
Update node version in readme (#16163)

Closes #16162

plabadie committed:
chore: Update the feature level. (#16116)

mono424 committed:
Feature/cxspa 836 (#16096)

* fixed pm2 issue

* re-run rebuild

Co-authored-by: Khadim Fall <khadim.fall@sap.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

justinlee01 committed:
Adding E2E test for checking that out of stock products do not have Add-to-Cart buttons for Product Carousel (#15975)

plabadie committed:
chore: Improve breaking change migration handling automation for major versions (#16076)

(CXSPA-823)

znikola committed:
docs: Update link for wrapper modules (#16174)

Updated schematics README to include the links to:

- wrapper schematics
- a guide about installing extensions for lazy loaded features

justinlee01 committed:
feat: The 'Save cart for later' button in <cx-add-to-saved-cart> is disabled if its current cart has no entries (#16021)

* The 'Save cart for later' button in <cx-add-to-saved-cart> is disabled if its current cart has no entries
* Changing styles so that the border happens after the 1st element, not only on the last child
* E2E test for disabling the Save Cart For Later button when there is no cart to save for later
Closes https://jira.tools.sap/browse/CXSPA-527

giancorderoortiz committed:
SonarQ fix. Modify variable declaration from 'let' to 'const' (#16008)

* Sonarq fix. Modify variable declaration from 'let' to 'const'

* update additional files with 'const' instead of 'let'

* force rebuild

* force rebuild

* force rebuild x2

Co-authored-by: Gilberto Alvarado <gilberto.alvarado.jimenez@sap.com>

bgambocjaviniar committed:
feat: Automate sample data publishing to ccv2 spartacus repo - githubactions + scripts (#16134)

closes https://jira.tools.sap/browse/CXSPA-851

FollowTheFlo committed:
fix: add XSS security for icon-loader (#16182)

Url sanitizer has been added for required url used by svg and link icon types.
Html symbol is now converted into text for text icon type.

RadhepS committed:
chore: Add release files to release branch (#16191)

Add release files to open source project

closes: https://jira.tools.sap/browse/CXSPA-1365

bgambocjaviniar committed:
chore: Copy code from public repository to private repository (#16110)

closes https://jira.tools.sap/browse/CXSPA-852

WeizhengSap committed:
feat: upgrade to Angular 14 (#16175)

Main changes:

    bump @angular/ packages from v13.3 to v14.2
    bump typescript from v4.6 to v4.7
    bump ng-packagr from v13.3 to v14
    bump relative packages (e.g. @nguniversal, @ngrx, ng-select etc.) to the highest version compatible with ng14
    upgrade some other 3rd party dependencies

dependabot[bot] committed:
chore(deps): bump terser from 5.12.1 to 5.14.2 (#16196)

* chore(deps): bump terser from 5.12.1 to 5.14.2

Bumps [terser](https://github.com/terser/terser) from 5.12.1 to 5.14.2.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/compare/v5.12.1...v5.14.2)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* force rebuild

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <giancarlo.cordero.ortiz@sap.com>

giancorderoortiz committed:
chore: Remove release-it and associated functionality

WeizhengSap committed:
chore: update webApplicationInjector.js (#16208)

znikola committed:
chore: Improve upgrade experience for v5  (#16184)

bgambocjaviniar committed:
chore: Update ng version installation in pipeline files (#16214)

bgambocjaviniar committed:
chore: Spartacus CCv2 repo update on demand (#16144)

closes https://jira.tools.sap/browse/CXSPA-1247

bgambocjaviniar committed:
chore: New CCv2 git configs (#16221)

RadhepS committed:
feat: automate the process of bumping package versions for publishing (#16212)

Included feature to bump package versions for publishing

closes: https://jira.tools.sap/browse/CXSPA-853

plabadie committed:
chore: Update the version in the core styles lib. (#16209)

FollowTheFlo committed:
fix: add xss security (#16225)

HTML escape is now applied to json-ld content.

ChristophHi committed:
fix: calculate configurator value prices after group navigation (#16244)

RadhepS committed:
feat: add canvas as optional peer dependency (#16249)

Canvas does not need to be installed since it is a peer dependency that is not used. This is the default behaviour for yarn but, we need to add this configuration to tell npm the same. RBSC uses npm.

closes: https://jira.tools.sap/browse/CXSPA-1479

ArkadiySukharenko committed:
Generate an error when invalid cost center is selected (#15826)

* Generate an error when invalid cost center is selected

* Fix unit test

* Fix

* Code review fixes

* Add E2E test to check for invalid cost center error

* Fix lint

* Fix lint

* Code review changes

* Fix unit test

* Code review fixes

FollowTheFlo committed:
fix: prevent two forms to be rendered at same time (#16256)

Fix for app frozen after logout during address edition in address book view.
A condition has been added to prevent 'add-address' and 'edit-address' forms to be rendered at the same time.

cmanciero committed:
feat: bind anonymous cart to a registered user in the ASM interface (#15776)

The new feature to bind an anonymous cart to an emulated user is available after starting a
customer emulation session.  A new component will appear in the ASM header that provides an input
and "Bind" button.  The input will by default display the current bound cart (if there is one) and allow
the assisted service agent to enter a new cart ID to bind using the "Bind" button.

New code includes a new facade/adapter/connector set for the cart bind operation, a new
loader component (dot-spinner) currently intended for use in ASM UI, and a new component for 
the ASM header to bind a cart.

A known issue is that the new styles for 5.1 are temporarily activated by default until a future fix to
style libraries that will make the base style version level work with feature libraries.

bgambocjaviniar committed:
chore: Renaming and improvements of released sample data (#16202)

closes https://jira.tools.sap/browse/CXSPA-1410

RadhepS committed:
feat: set productiveBranch when copying from public to private repo (#16253)

Set the productiveBranch value to the name of the copied branch when copying from public to private repository

closes: https://jira.tools.sap/browse/CXSPA-1483

pawelfras committed:
NgBootstrap removal develop backport

pawelfras committed:
bugfix: CXSPA-1486 Added To Cart Dialog not closing after clicking outside

Larisa-Staroverova committed:
feat: Enable expert mode (#16034)

Display technical group IDs, attribute IDs and attribute domain values, KB information in the header section of the interactive configuration page

Closes CXSPA-857

kuhris-sap committed:
Feature CXSPA-336 - Account Summary (#16081)

* account-summary init

* enable occ module + cleanup

* clean up

* Fix branch for production build

* Changes for automated deployment (REVERT BEFORE DEVELOP MERGE)

* delete generated css files

* Dummy change to test deployment script

* wip - remove css references + display header data

* remove ag-grid dependency

* fix unit list translations + rename details routing path + add basic scss index

* breadcrumbs files + refactor cms config

* feature/CXSPA-336: add AccoountSummaryPageMetaResolver

* refactor account summary page meta resolver

* refactor cms configs + add other style index files

* map fields

* Add boolean for Add button in ListComponent

* Update model and translation

* fix conflicts + add core + converters + facades + service (wip)

* put back details directory + move page resolver

* update cms component name

* fix components index paths

* account summary details endpoint

* add document list endpoint + DocumentQueryParams

* Update Header and Document components

* css fixes to account summary details page

* extend unit list component

* Convert header details to CSS Grid

* Update OCC endpoints

* Fix alignment of documents table

* Add pagination

* Add sorting

* fix imports + prod build errors - wip

* fix renamed files

* fix build

* add account summary details page auth guard

* get it working

* fix layout

* more fixes

* css fixes

* refactor filters

* refactor filter css

* fix action items from UX review

Text changes, removing duplicate header in documents

* fix build

* Convert data request Observable to BehaviorSubject

* add caching on service layer

* modify text

* no longer need ngrx store

* Add attachment icon

* Add some error handling

* replace formatted currency fields with local formatter

* minor fixes

- Update OCC endpoint for header details
- Change icons from <i> to cx-icon
- Add tooltips to icons
- Cache document types + sort options from document request and then subsequent requests ask for BASIC data
- Rename css class to use naming convention
- Add comments

* update error messages

* lint errors

* Adapt header detail to API changes

* fixes for style code review

* more style fixes

* remove cell link component

* rename account-summary.ts to account-summary.model.ts

* move occ endpoint models to its own file

* add unit tests - wip

* add spartacus styles dependency

* css lint fixes

* update account summary document endpoint

* update mapping to documents occ endpoint

* some accessibility fixes

* prettier fixes (excluding spec)

* fix ui mapping

* fix merge conflicts + add unit tests

* prettier on spec files

* fix hardcoded string

* fix build

* fix lint issues

* account-summary-document tests

* account-summary-header tests

* account-summary-filter tests

* test

* add file conversion to blob + download file

* Add test coverage

* add endpoint error handling

* refactor unitCode to orgUnit

* rename account summary endpoint

* add attachment unit tests + change hard coded type

* adapt to new attachments field in documents enpoint

* add file name to attachment

* add attachment download file test

* update attachments field in mock data

* fix lint

* remove test focus

* clean up attachments

* add schematics config

* rename feature name

* fix schematics test + move feature module

* test feature repo change

* 1.0.0

* Revert "test feature repo change"

This reverts commit a95545366bb4ba7b419b3bb943da353540afe7b5.

* Revert "1.0.0"

This reverts commit 46551e3920a4259234afad2f86a5249774953c38.

* update schema.json

* rename prompt label

* fix format

* clean up

* clean up test

* fix accessibility for attachment download

* review fixes (first pass)

* generate snapshot

* review fixes + some refactor

* static code check

* re-add lost commit changes

* convert queryParams to rxjs Subject

* rename inconsistent sort and filter names

* Make Spartacus Language selector set currency locale

* Incomplete filter changes not reset when changing sort

* Prefer pre-formatted currency when available

* review fix (2nd pass)

* Update README.md

* bugFix: CXSPA-1390 - Language switcher

* fix accelerator inconsistencies

* fix unit test

* code review changes part 1

* fix unit test expectation

* code review changes part 2

* code review changes part 3

* code review changes part 4

* code review changes part 5

* code review changes part 6

* Trigger ci

* Update package.json

* move augmented model from root to component folder

* code review changes part 7

* Trigger ci

* code review changes part 8

Co-authored-by: Sam Kazerouni <sam.kazerouni@sap.com>
Co-authored-by: i829560 <ronald.santiago@sap.com>
Co-authored-by: Weizheng Gao <weizheng.gao@sap.com>

RadhepS committed:
feat: automatically trigger azure pipeline build when copying code to private repository (#16282)

closes: https://jira.tools.sap/browse/CXSPA-1511

RadhepS committed:
test: fix failing my-coupons e2e 

Fixes my-coupons e2e test failing

closes: https://jira.tools.sap/browse/CXSPA-1512

FollowTheFlo committed:
chore: add capability to install spartacus from private packages (#16213)

Spartacus can now be installed from sap private npm packages when using install_npm command.
Config vars have been added: NPM_TOKEN, NPM_URL,NPM_ALWAYS_AUTH
When the install starts, .npmrc file gets created then removed once finished.
For security purpose, NPM_TOKEN var gets cleared from config.sh and config.default.sh when install is finished.

BREAKING CHANGE

New config vars have been introduced: NPM_TOKEN, NPM_URL are required to be filled by the user in config.sh file. It is needed only when using install_npm command.

justinlee01 committed:
feat: Customer ID Http Header interceptor for ASM emulation (#16095)

Closes CXSPA-574

* Creating UserIdInterceptor, that appends sap-commerce-cloud-user-id header to requests for certain endpoints when a customer is emulated

* No longer using glob patterns to identify which endpoints need the sap-commerce-cloud-user-id header - instead, using HttpContext for certain endpoints; the UserIdInterceptor looks for a specific context to know which endpoints to send the header with.

* Adding ASM B2B test using costcenters. When listening for user details in B2B, listen for /orgUsers/ endpoint. Using internal interceptGet for addProductToCart helper

* Moving feature toggle for enabling sap-commerce-cloud-user-id header to AsmConfig, which is more module-appropriate

* Copying AsmConfig and deprecating the old one, as the configuration should be in /root instead of /core

* Adding ticket number for currently skipped tests awaiting backend changes

Platonn committed:
chore: add comment why we can't inject Router via deps[] for addCmsRoute factory (#16286)

FollowTheFlo committed:
fix: add html escape to prevent xss attacks (#16242)

HTML escape is now applied to json-ld content to prevent xss attacks.

kimhw0630 committed:
Feature/cxspa 206 (#15702)

* first commit for ASM

Co-authored-by: Eric Polesky <eric@unboundcommerce.com>

* Co-authored-by: Eric Polesky <eric@unboundcommerce.com>

* Fiori

* implement asm query service

* add customer search to asm facade

- move more of model into root to support new facade intnerface

* remove fiori and ngrx for customer group lists

* test sort

* update design

* add beta parameter support to query system

* use modified query with parameters and include additional ui updates

* pagenation

* fix unit tests

* added spinner while waiting for backend

* unit tests for customer list & main ui components

* some style changes

* missing localization

* e2e basic

* updated sorting to handle dynamically

* remove unused property

* rollback yarn lock file

* update peerDependencies

* fix for css property-disallowed-list

* re-generate dependencies.json

* fix asm/root circular dependency with self

* UX change for list

* column title change for email

* response design

* update breakpoint

* Revert "add beta parameter support to query system"

This reverts commit 3e61c09b7859f927c289d7b8e3ccf5a81992d99a.

* remove beta query usage

* update unit tests

* fix prettier format

* add unit tests for asm-query service

* accessibility table cell

* use cxFocus

* updated e2e test

* configuration for asm customer list

Co-authored-by: Eric Polesky <ercultimate@users.noreply.github.com>

* fix build issue with using protected property

* fix avatar position

* fix accessibility for ng-select

* fix build error

* replaced with cx-sort

* fix accessibility issue with dialog

* fix unit test for asm main

* rename style class name to follow cx-naming

* fix accessibility issue in sort

* fix for style code review

* code review style fix

* added feature flag

* update unit test

* change for code review

* remove breaking changes from customer list feature

- move migrated models back to original location
- remove facade methods to prevent cyclical dependency cuased by
  depending on asm/core in asm/root module
- implement replacement ngrx state and selectors for removed facade method

* rename asm-facade and asm-query service to more appropriate values

* remove barrel from root of asm/root module

* update import paths for unmigrated models

* fix switched cypress helper interceptor aliases

* fix misordered constructor arguments

* clean up code

* update unit tests for ngrx customer list customers search

* export a service

* fix lint error

* fix breaking change in asm service constructor

* Add missing license header to sources

* update public_api.ts for customer list

* remove local angular cli settings

* move getCustomerLists to customer list facade

* normaize errors for customer list request

* reorder adapter conversion and error normalization to match convention

* fix duplicate facadeProvider spread from merge

* clean up development code notes

* add unit tests for asm connector bind cart method

* replaced ModalService with LaunchDialogService

* code review changes

* code review: removed extend OccConfig

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>
Co-authored-by: Eric Polesky <eric@unboundcommerce.com>
Co-authored-by: Eric Polesky <eric.polesky@sap.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>
Co-authored-by: Eric Polesky <ercultimate@users.noreply.github.com>

mono424 committed:
Fix: Cannot change language and currency (#16258)

* fix issue

* remove default for language & currency

* fix tests

* fix lowercase & fix tests

Co-authored-by: Khadim Fall <khadim.fall@sap.com>

ercultimate committed:
chore: change bind cart service to use uid as customer identifier (#16293)

The proper value to be passed into the bind cart call's customerId parameter is the customer UID.  This patch changes the bind cart call from using the customerId value to using the uid value.

Co-authored-by: Eric Polesky <eric.polesky@sap.com>

kuhris-sap committed:
Add copyright headers to Account Summary (#16303)

* Add license header to sources

* rename to orgUnitId

ChristophHi committed:
feat: request configurator pricing on demand (#16304)

With commerce 22.11, OCC can signal that pricing is supported. If this is not the case, we will not do the pricing calls to the backend.
For older commerce releases, nothing changes, pricing is always requested.

ChristophHi committed:
feat: display value prices for cart configurations (#16280)

Value prices are displayed when a cart bound configuration is launched. No further interaction is needed like before

ercultimate committed:
CXSPA-1299: Update bind cart e2e (#16302)

* chore: update bind cart e2e to cover full stated test case

* force rebuild

* test rerun

* force rebuild

Co-authored-by: Eric Polesky <eric.polesky@sap.com>
Co-authored-by: Giancarlo Cordero Ortiz <giancarlo.cordero.ortiz@sap.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

ChristophHi committed:
chore: all cpq attributes are visible (#16308)

Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

ChristophHi committed:
chore: fix strict mode issues / configurator tests (#16306)

Zeyber committed:
fix: Change provideConfig to provideDefaultConfig in modules and add lint (#16287)

Closes: https://jira.tools.sap/browse/CXSPA-289

mono424 committed:
chore: improve install script

Platonn committed:
fix: adapt configuration related additions in cart to refactored tables (#16307)

bundle-info component: style always in a compact way
issues-notification component: reduce the margin-top

closes #15566

Larisa-Staroverova committed:
fix: Adjust styling for image at attribute level (#16320)

Adjust styling for image at attribute level

Closes CXSPA-913

mono424 committed:
chore: Installation Script Fixes & Improvements (#16170)

SherwinVarghese committed:
feat: CDC-COMM:SPA UI - Headless Commerce: Login & Registration

bgambocjaviniar committed:
fix: Message after registering a new user is not displayed anymore (#16335)

closes https://jira.tools.sap/browse/CXSPA-1557

bgambocjaviniar committed:
chore: Automate the prepended licensed header using github actions for the public repository (#16243)

closes https://jira.tools.sap/browse/CXSPA-1083

bgambocjaviniar committed:
feat: Support for better specificity on page meta resolvers (#15985)

Gave more weight to when pageUid is taken into consideration. Kept the logic isolated as such with the resolvers and did not leak it to the applicable and page meta service. Basically, the scoring logic should be isolated within the resolvers as the current logic is iterating through the list of resolvers and matching the page *resolver that corresponds to the current page

closes https://jira.tools.sap/browse/CXSPA-757

ChristophHi committed:
chore: fix strict mode issues in configurator tests (#16322)

Zeyber committed:
chore: Fix flaky b2b my company e2e tests (#16331)

Closes: https://jira.tools.sap/browse/CXSPA-1533

AlexanderKuschner committed:
feat: enable VC configurator for ASM

ChristophHi committed:
feat: send actual b2b user for cpq token request (#16346)

In order to support ASM

ChristophHi committed:
fix: enable pricing for CPQ configurator (#16351)

hackergil committed:
chore: github action naming changes

Co-authored-by: Brian Gamboc-Javiniar <brian_javiniar@hotmail.com>

znikola committed:
chore: Remove changelog script and related dependencies (#16343)

closes CXSPA-892

giancorderoortiz committed:
Temporarily disable asm tests due to high failure rate (#16318)

* Temporarily disable asm tests due to high failure rate

* force rebuild

* force rebuild 2

* force rebuild 2

* force rebuild 3

plabadie committed:
chore: Update Code Owners (#16349)

bgambocjaviniar committed:
chore: Run entire E2E suite on demand for testing purposes (#16360)

closes https://jira.tools.sap/browse/CXSPA-1588

bgambocjaviniar committed:
chore: Running tests on CCv2 that are running unreleased Spartacus versions (#16372)

closes https://jira.tools.sap/browse/CXSPA-1603

bgambocjaviniar committed:
chore: Filter ccv2 e2e folder for e2e tests (#16376)

exclude cx_ccv2 tests from being run

bgambocjaviniar committed:
chore: Fix cart / cart-core e2es (#16371)

closes https://jira.tools.sap/browse/CXSPA-1587

kimhw0630 committed:
updated customer list styling according mocks (#16378)

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
fixed AMP violation on saved cart details page (#16379)

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

ishrarhussain committed:
CXEC-20507: Add Cypress E2E scenarios for viewing ticket details via URL

ishrarhussain committed:
CXEC-20507: Add Cypress E2E scenarios for viewing ticket details via URL

bgambocjaviniar committed:
chore: Remote duplicate ccv2 tests and specify them instead (#16386)

closes https://jira.tools.sap/browse/CXSPA-1617

ishrarhussain committed:
Update cutsomer-ticketing.ts

ishrarhussain committed:
Update cutsomer-ticketing.ts

adolfobee commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/cutsomer-ticketing.ts:
```diff
@@ -40,3 +40,20 @@ export function verifyTicketDetailsPageVisit(){
 }
+
+export function visitTicketDetailsPageForFirstTicket(){
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).should('exist');
```
can we have the .should in the next line? it seems like a pretty long line

khristian23 replied:
Is this line really needed?
If the item does not exist in the DOM, the following line (46) will fail.

adolfobee replied:
I just want the legth of the line to be shorter!

osahtout replied:
I think it can just be 
`cy.get('cx-customer-ticketing-list').get('tbody').get('tr').eq(FIRST_ROW_TICKET_LIST);`

ishrarhussain replied:
Thanks guys. I think the CSS selector works for efficiently. Also, the use of `tbody` here is to differentiate its downstream `tr`s from the ones that may be in `thead`.

adolfobee commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/cutsomer-ticketing.ts:
```diff
@@ -40,3 +40,20 @@ export function verifyTicketDetailsPageVisit(){
+
+export function visitTicketDetailsPageForFirstTicket(){
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).should('exist');
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).then( ($ticketIdElement) => {
```
same for .then

khristian23 replied:
Since this is a CSS-like selector tbody should not be needed.

FIRST_TICKET_ROW_INDEX would be changed by "ticketPosition" passed as parameter.

FIRST_TICKET_COLUMN_INDEX what value does it contain? How about using the name of the column?  Something like COLUMN_SUBJECT.  We don't need prefix FIRST_TICKET

osahtout replied:
I have some of these defined in the merged e2e and some in the ticket list e2e

ishrarhussain replied:
They are now fixed. Please check and confirm.

adolfobee commented:
Left some comments that might require prettify, make sure lines length is no longer than 80/100 columns max

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/cutsomer-ticketing.ts:
```diff
@@ -40,3 +40,20 @@ export function verifyTicketDetailsPageVisit(){
   cy.get('cx-customer-ticketing-messages').should('exist');
 }
+
+export function visitTicketDetailsPageForFirstTicket(){
```
How about parametrizing this method to accept a ticket position in the table.
Something like:
```suggestion
export function visitTicketDetailsPageForTicketInPosition(ticketPosition: number){
```

Like this, we can reuse the same method when we to access the a ticket in the second, third... n position

ishrarhussain replied:
Done.

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/cutsomer-ticketing.ts:
```diff
@@ -40,3 +40,20 @@ export function verifyTicketDetailsPageVisit(){
+    const ticketId = $ticketIdElement.text().trim();
+    cy.visit(`/my-account/support-ticket/${ticketId}`);
+    cy.get('cx-customer-ticketing-messages', { timeout: 10000 }).should('be.visible');
+    cy.url().should('include',`/my-account/support-ticket/${ticketId}`);
```
```suggestion
    visitTicketDetailsPageForTicketId(ticketId);
```
Please see next suggestion

ishrarhussain replied:
These should all now be addressed in the newly refactored version. @khristian23 Please check and confirm.

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/integration/regression/customer-ticketing/view-ticket-details.e2e-spec.ts:
```diff
@@ -17,6 +17,21 @@ describe('ticketing', () => {
+        customerTicketing.clickMyAccountMenuOption();
+        customerTicketing.clickCustomerSupportMenuOption();
+        customerTicketing.verifyTicketListingPageVisit();
+        // Uncomment the following action after 404 error flow is merged on epic
+        // customerTicketing.visitTicketDetailsPageForNonExistingTicket();
```
It is OK if it fails, that will force us to provide the fix :) PLease uncomment

khristian23 replied:
```suggestion
        customerTicketing.visitTicketDetailsPageForTicketId(NOT_EXISTENT_ID)
        customerTicketing.verifyTicketListingPageVisit();
        customerTicketing.verifyErrorMessageIsShownWithText(TICKET_NOT_FOUND);
```

ishrarhussain replied:
Done.

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/cutsomer-ticketing.ts:
```diff
@@ -40,3 +40,20 @@ export function verifyTicketDetailsPageVisit(){
+  });
+}
+
+export function visitTicketDetailsPageForNonExistingTicket(){
+  cy.visit('/my-account/support-ticket/XYZ01234');
+  cy.get('cx-global-message .alert-danger', { timeout: 10000 }).should('be.visible');
+  cy.get('cx-global-message .alert-danger span').should('include.text', 'Ticket not found.');
+  cy.url().should('include','/my-account/support-ticket');
+}
```
Maybe you can split "visitTicketDetailsPageForNonExistingTicket" function into two.
On the one hand, you can create a generic method like:
```
export function visitTicketDetailsPageForTicketId(ticketId: string) {
  const ticketDetailsPageUrl = `/my-account/support-ticket/${ticketId}`;
  cy.intercept(ticketDetailsPageUrl).as('ticketDetails');
  cy.visit(ticketDetailsPageUrl);
  cy.wait('@ticketDetails');
}
```

Like this we can have:
const NOT_EXISTENT_ID = 'XYZ01234';
customerTicketing.visitTicketDetailsPageForTicketId(NOT_EXISTENT_ID);

The second will check the existence of a banner on a page:
```
export function verifyErrorMessageIsShownWithText(errorMessage: string) {
  const TEN_SECONDS = 10000; // But Why ten seconds ?? Isn't that too much?
  cy.get('cx-global-message .alert-danger', { timeout: TEN_SECOND }).should('be.visible');
  cy.get('cx-global-message .alert-danger span').should('include.text', errorMessage);
}
```

khristian23 replied:
Moreover, there is now a function that does this:
https://github.com/SAP/spartacus/blob/f3761fa13e9541123d98e0bb1a46e6e44bdd5b8b/projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts#L60

So you will need to do:
```
export function visitTicketDetailsPageForTicketId(ticketId: string) {
  const ticketDetailsPageUrl = `/my-account/support-ticket/${ticketId}`;
  visitPage(ticketDetailsPageUrl);
}
 ```

ishrarhussain replied:
@khristian23 Refactored it as per this request. Please check and confirm. Thank you so much!

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/integration/regression/customer-ticketing/view-ticket-details.e2e-spec.ts:
```diff
@@ -17,6 +17,21 @@ describe('ticketing', () => {
+        customerTicketing.clickMyAccountMenuOption();
+        customerTicketing.clickCustomerSupportMenuOption();
+        customerTicketing.verifyTicketListingPageVisit();
+        customerTicketing.visitTicketDetailsPageForFirstTicket();
```
```suggestion
        const FIRST_TICKET = 0;
        const TICKET_NOT_FOUND = "Ticket not found";
        customerTicketing.visitTicketDetailsPageForTicketInPosition(FIRST_TICKET);
        customerTicketing.verifyTicketDetailsPageVisit();
```

ishrarhussain replied:
It's now done.

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/integration/regression/customer-ticketing/view-ticket-details.e2e-spec.ts:
```diff
@@ -17,6 +17,21 @@ describe('ticketing', () => {
+      });
+      it('should throw 404 error when trying to visit ticket details page for a non-existing ticket id via url', () => {
+        customerTicketing.loginRegisteredUser();
+        customerTicketing.clickMyAccountMenuOption();
+        customerTicketing.clickCustomerSupportMenuOption();
+        customerTicketing.verifyTicketListingPageVisit();
```
I guess we don't need these lines since we are navigating by URL to ticket details page

khristian23 replied:
Just discussed this with Omar, and you might need to navigate to the ticket listing first.
But try without those three lines and see if it works

ishrarhussain replied:
Done.

kimhw0630 committed:
Feature/cxspa 303 (#16139)

* enabled scrollPositionRestoration for improve scroll restoration

* Co-authored-by: Eric Polesky <ercultimate@users.noreply.github.com>

* updated e2e due to timeout

* update unit test

* add a feature flag and comment

* Added deprecated annotation to searchable when upgrading to major

* added notes in 6_0.md

* typo in changelog

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
feature: accessibility issue in coupon page (CXSPA-1435) (#16384)

* fixed accessibility issue in coupon page

* replaced anchor to button to tabbing correctly

* added comments for breaking changes

* updated changelog

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

ishrarhussain committed:
Merge branch 'epic/customer-ticketing' into task/CXEC-20507-cypress-e2e-scenarios-for-ticket-details-by-url

ishrarhussain committed:
Merge branch 'epic/customer-ticketing' into task/CXEC-20507-cypress-e2e-scenarios-for-ticket-details-by-url

osahtout commented:
Just make sure to correct the name of the file from `cutsomer` to `customer`

Platonn committed:
chore: Use "development" configuration for "yarn dev:ssr"  (#16417)

Now for `yarn dev:ssr` we use the "development" configuration in `angular.json`, which generates sourcemaps and disables optimizations (e.g. minification) of code. Thanks to that now it's much easier to debug SSR, while running `yarn dev:ssr`. Now when you `Attach to a Node process` (to Node process of `yarn dev:ssr`) e.g. in VSCode and set a breakpoint for a lines in the source code, the control is paused at this line, when SSR request is processed.

closes CXSPA-1672

ishrarhussain committed:
Merge branch 'epic/customer-ticketing' into task/CXEC-20507-cypress-e2e-scenarios-for-ticket-details-by-url

ishrarhussain committed:
Delete cutsomer-ticketing.ts

ishrarhussain committed:
Delete cutsomer-ticketing.ts

ishrarhussain committed:
Updates to view ticket details e2e tests as per feedback

ishrarhussain committed:
Updates to view ticket details e2e tests as per feedback

ishrarhussain commented:
> Just make sure to correct the name of the file from `cutsomer` to `customer`

Right, I saw it getting inherited initially from the helper file of your branch... It's now removed.

ishrarhussain commented:
Thanks all! ..All the requested changes have now addressed. Please review @khristian23 , @adolfobee , @osahtout

ishrarhussain committed:
Update view-ticket-details.e2e-spec.ts

ishrarhussain committed:
Update view-ticket-details.e2e-spec.ts

ishrarhussain committed:
Some more changes as per PR review

ishrarhussain committed:
Some more changes as per PR review

ishrarhussain committed:
Update customer-ticketing.ts

ishrarhussain committed:
Update customer-ticketing.ts

ishrarhussain committed:
Another minor update as per feedback

ishrarhussain committed:
Another minor update as per feedback

justinlee01 committed:
test: Updating scroll-position-restoration tests to not be data-dependent (CXSPA-1662) (#16409)

- no longer asserting for specific products
- no longer using hardcoded url

ishrarhussain committed:
Merge branch 'epic/customer-ticketing' into task/CXEC-20507-cypress-e2e-scenarios-for-ticket-details-by-url

bgambocjaviniar committed:
chore: Stabilize infinite scroll e2e (#16413)

closes https://jira.tools.sap/browse/CXSPA-1663

osahtout commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts:
```diff
@@ -48,8 +49,8 @@ export function verifyTicketListingPageVisit(){
 
-export function clickFirstTicketFromTicketListing(){
-  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).click();
+export function clickTicketFromTicketListing(ticketRowIndex: number){
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${ticketRowIndex}) .cx-ticketing-list-data:nth-child(${TICKET_ID_COLUMN_INDEX}) a.cx-ticketing-list-value`).click();
```
I think `cy.get('cx-customer-ticketing-list').get('tbody').get('tr').eq(ticketRowIndex).click();` might be simpler. Used [here](https://github.com/SAP/spartacus/blob/48fb4363e7a3f172d852c76e39d01ac6098b8b7f/projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts#L124)

osahtout commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts:
```diff
@@ -149,3 +149,25 @@ export function verifyTicketDoesNotExist(ticketDetails: TestTicketDetails) {
+  visitPage(ticketDetailsPageUrl);
+}
+
+export function visitTicketDetailsPageForTicketInPosition(ticketRowIndex: number){
```
I'm not sure why this function is needed

khristian23 replied:
So, we have some functions named something like: VIsit Ticket Details Page for First Ticket Listed
The idea is to stop using a function for FIRST, SECOND etc and make a generic function that received the line number you want to navigate to

osahtout replied:
I understand the idea and I do need it for sorting but in this PR we have clickTicketFromTicketListing and visitTicketDetailsPageForTicketInPosition

khristian23 replied:
Oh my bad, in that case, we need to choose one.
My personal preference is "click..." because the "visit..."  could mean we are visiting through the URL

steinsebastian committed:
style: Fix several styling issues (#16381)

The following styling issues are fixed:
* CXSPA-896: Fix line break
* CXSPA-896: Fix styles for conflict groups for intervals
* CXSPA-896: Fix suggestion box style
* CXSPA-896: Fix conflict navigation link style

ChristophHi committed:
feat: enable expert mode for cart configurations (#16431)

adolfobee committed:
Addressing accesibility pointers (#16427)

Co-authored-by: Adolfo <adolfo.pereira@sap.com>

ishrarhussain committed:
CXEC-17550: Navigate to Ticket Listing in case of 404 Not Found Ticket Detail (#16405)

* CXEC-17550: Navigate to Ticket Listing in case of 404 Not Found Ticket Detail

Jira Issue # CXEC-17550 (https://jira.tools.sap/browse/CXEC-17550)

* Simplify error details retrieval with constants

+ Added separate module for constants in error details strings

Co-Authored-By: Christian Montoya <christian.montoya.b@gmail.com>

* Added check on http status code

* Added a new http-interceptor as per request on the PR review

Co-authored-by: Christian Montoya <christian.montoya.b@gmail.com>

khristian23 commented on projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts:
```diff
@@ -149,3 +149,25 @@ export function verifyTicketDoesNotExist(ticketDetails: TestTicketDetails) {
+  cy.get('cx-global-message .alert-danger', { timeout: 10000 }).should('be.visible');
+  cy.get('cx-global-message .alert-danger span').should('include.text', TICKET_NOT_FOUND);
+  cy.url().should('include','/my-account/support-ticket');
+}
```
I guess this method could be split into these:

visitDetailsPageForTicketId(NON_EXISTENT_TICKET)
verifyErrorMessageWithTextIsSeen(TICKET_NOT_FOUND)
verifyCurrentPageIsTicketListingPage()

ChristophHi committed:
test: fix conflict solver e2e test

kimhw0630 committed:
feature: Accessibility Heading level on add/edit address (CXSPA-1107) (#16390)

* feature: Accessibility - Heading level on address book

* added comments for breaking changes

* added missing bottom margin

* added TODO notes for breaking change

* used forVersion on style change

* updated changed log

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

FollowTheFlo committed:
test: b2b user creation has been refactored for cx 2211 compatibility (#16397)

Two changes have been done to make e2e tests compatible with cx 2211:
B2b users are now created by admin so the approval required by cx 2211 is bypassed.
B2b user delivery address id is now fetched from server rather than hard-coded in mock object.

adolfobee committed:
feature/CXSPA-1685-1683 - Modify style of ticket list table (#16439)

* Modify style of ticket list table

* Lint

Co-authored-by: Adolfo <adolfo.pereira@sap.com>

Parthlakhani committed:
CXEC-20553: fix scrolling on load and send message (#16424)

Parthlakhani committed:
Merge branch 'develop' of https://github.com/SAP/spartacus into epic/customer-ticketing

github-actions[bot] committed:
Add license header

RadhepS committed:
chore: remove Yotpo

Removes source code associated with Yotpo

closes: https://jira.tools.sap/browse/CXSPA-1112

dependabot[bot] committed:
chore(deps): bump loader-utils from 2.0.2 to 2.0.3 (#16447)

* chore(deps): bump loader-utils from 2.0.2 to 2.0.3

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 2.0.2 to 2.0.3.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v2.0.3/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v2.0.2...v2.0.3)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* force rerun

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <giancarlo.cordero.ortiz@sap.com>

giancorderoortiz committed:
Temporarily disable cart import tests due to flakiness and failures in pipeline (#16461)

osahtout committed:
Refactor customer ticketing e2e helper (#16457)

* CXEC-15927 implement ticket listing e2e

* refactor-customer-ticketing-e2e-helper

plabadie committed:
chore: codeowners improvements (#16462)

Uli-Tiger committed:
feat: hide selected and options price from price summary if requested (CXSPA-1527) (#16455)

In case the backoffice setting is exposed via OCC (from commerce 2211.1 onwards), base and option price will be dynamically shown/hidden based on the setting. 
In case the setting is not returned by OCC (older commerce versions), the options and base price will be shown as before.

dependabot[bot] committed:
chore(deps): bump loader-utils in /projects/storefrontapp-e2e-cypress (#16460)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.0 to 1.4.1.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.1/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.0...v1.4.1)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

giancorderoortiz committed:
test: disable all product configurator e2e tests pending cx upgrade (#16469)

All product configurator e2e tests have been disabled due random failures towards CI 2211 version. 
Investigation is needed until its reactivation.

Parthlakhani committed:
CXEC-17824/16883: restructure customer ticketing and resolve ticketing details header (#16465)

RadhepS committed:
test: fix anonymous-consents config e2e

WeizhengSap committed:
chore: fix flaky group skipping core e2e (#16404)

kimhw0630 committed:
feature: accessibility improvements on Saved Carts & Payment Details headers(CXSPA-1106) (#16391)

* Replaced h3 tag with h2 tag for accessibility improvements

* updated changed log

* updated todo commands

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
feature: accessibility improvements on my coupons &  interests (CXSPA-1105) (#16393)

* feature: accessibility improvements on my coupons & interests

* updated changelog 6_0.md

* updated change log

* update todo comments

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
feature: accessibility improvements on quick order list header (CXSPA-1093) (#16434)

* replaced header tag for accessibility improvements

* updated todo comment

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

Uli-Tiger committed:
fix: icons in group menu shall have a tooltip (CXEC-16787) (#16479)

* fix: icons in group menu shall have a tooltip (CXEC-16787)

bgambocjaviniar committed:
chore: Minimize number of e2es being run on ccv2 (#16481)

bgambocjaviniar committed:
feat: Invalidate user upon password change (#16472)

closes https://jira.tools.sap/browse/CXSPA-1242

steinsebastian committed:
fix: long text for exit button for screen readers (#16483)

Adding title attribute with long text "Exit Configuration" or "Cancel Configuration".
Thus a tooltip is generated and the tooltip information is read by the screenreader

Parthlakhani committed:
CXEC-16053: add unit tests (#16488)

bgambocjaviniar committed:
fix: Cart items not matching the number of actual imported items to a saved cart (#16486)

* stop queuing in case and use recent emissions only

* fix second part, where it flaked given missing products that were supposed to be exported

* make sure clean 'slate' for every run just in case

* don't clear everything per assertion - made it flaky haha..

* checkout for when cart has been 'updated' when button is visible instead of the text as the text can be different

* slow down the test before next assertion (more robust - it wasn't flaky)

* trigger build

* trigger build 2

Platonn committed:
feat: make extendable `DEFAULT_SEARCH_CONFIG` of `OccProductSearchAdapter`

fixes CXSPA-1748

kimhw0630 committed:
fix: removed invalid aria role (CXSPA-1593) (#16470)

* removed invalid aria role

* added feature flag

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

Parthlakhani committed:
CXEC-20931: fix data reloading issue (#16493)

kimhw0630 committed:
separated button from header (#16451)

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
fix: accessibility on order history table header (CXSPA-1599) (#16467)

* accessibility on order history table header

* remove duplicate

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
AMP violations: fix id duplicate (#16484)

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

kimhw0630 committed:
removed aria-hidden attribute from active element (#16476)

Co-authored-by: Hakwoo Kim <hakwoo.kim@sap.com>

Parthlakhani committed:
CXEC-17553: add spinner until data is loaded (#16499)

dependabot[bot] committed:
chore(deps): bump loader-utils from 2.0.3 to 2.0.4 (#16485)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 2.0.3 to 2.0.4.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v2.0.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v2.0.3...v2.0.4)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump loader-utils in /projects/storefrontapp-e2e-cypress (#16502)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.1 to 1.4.2.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.2/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.1...v1.4.2)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

Parthlakhani committed:
Merge branch 'develop' into epic/customer-ticketing

Parthlakhani committed:
CXEC-21318: fix breadcrumbs (#16504)

manpreetlamba4 committed:
Add code in associated object list (#16521)

manpreetlamba4 committed:
CXEC-15126 Add unit tests (#16522)

* Add unit tests

* add tests

sushant-stha committed:
renaming file

sushant-stha committed:
Refactoring to reflect to Omars changes

sushant-stha commented:
Replaced with https://github.com/SAP/spartacus/pull/16541
